### PR TITLE
[FLINK-28437] Document Engine Matrix and Build page

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: docs/content/docs/engines/build.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+You can find Flink 1.15 bundled jar in `./flink-table-store-dist/target/flink-table-store-dist-{{< version >}}.jar`.
+
+## Hive
+
+You can find Hive bundled jar in `./flink-table-store-hive/flink-table-store-hive-connector/target/flink-table-store-hive-connector-{{< version >}}.jar`.
+
+## Spark
+
+You can find Spark bundled jar in `./flink-table-store-spark/target/flink-table-store-spark-{{< version >}}.jar`.
```
[the remark this answers is not in the document]
Comment (JingsongLi) [reply docs/content/docs/engines/build.md:53]: I think a full page to explain all engines is OK.

File: docs/content/docs/engines/overview.md [hunk 2]
```diff
@@ -29,3 +29,16 @@ under the License.
 Table Store not only supports Flink SQL writes and queries natively,
 but also provides queries from other popular engines, such as
 Apache Hive and Apache Spark.
+
+## Engine Matrix
```
Comment (LadyForest) [inline docs/content/docs/engines/overview.md:33]: How about `Version Compatibility Matrix` or `Compatibility Matrix`. It seems like a specific terminology, and I saw many docs using this term. 

E.g.
https://docs.docker.com/compose/compose-file/compose-versioning/
https://knowledge.autodesk.com/support/autocad/getting-started/caas/simplecontent/content/autocad-version-compatability-matrix.html

File: docs/content/docs/try-table-store/quick-start.md [hunk 3]
```diff
@@ -48,24 +48,7 @@ tar -xzf flink-*.tgz
...
-
-The simplest way of building Table Store is by running:
-
-```bash
-mvn clean install -DskipTests
-```
-
-You can find the bundle jar in `"./flink-table-store-dist/target/flink-table-store-dist-{{< version >}}.jar"`.
-
+You are using an unreleased version of Table Store, you need to manually [Build Bundled Jar]({{< ref "docs/engines/build" >}}) from the source code.
```
Comment (LadyForest) [inline docs/content/docs/try-table-store/quick-start.md:51]:
```suggestion
If you are using an unreleased version of Table Store, you need to manually [Build Bundled Jar]({{< ref "docs/engines/build" >}}) from the source code.
```

File: docs/content/docs/engines/trino.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Trino
+
+Because Trino's dependency is JDK 11, it is not possible to include the trino connector in [flink-table-store](https://github.com/apache/flink-table-store).
+
+See [flink-table-store-trino](https://github.com/JingsongLi/flink-table-store-trino).
```
Comment (LadyForest) [inline docs/content/docs/engines/trino.md:31]: It refers to a personal repo, and I'm unsure about the license-related stuff. Is it ok?

**Review by LadyForest — COMMENTED** [review]
Thanks for your contribution! It looks good to me in general, and I just left some minor comments.

File: docs/content/docs/try-table-store/quick-start.md (hunk 3 above)
Comment (JingsongLi) [reply docs/content/docs/try-table-store/quick-start.md:51]: `{{< unstable >}}`, it is a unreleased version.

File: docs/content/docs/engines/trino.md (hunk 4 above)
Comment (JingsongLi) [reply docs/content/docs/engines/trino.md:31]: It is a link instead of dependency. It is OK.

**Pull request merged**